core/new-gui/src/app/common/service/user/workflow-persist/workflow-persist.service.ts

```diff
@@ -28,6 +28,28 @@ export class WorkflowPersistService {
       .pipe(map(WorkflowPersistService.parseWorkflowInfo));
   }
 
+  /**
+   * creates a workflow and insert it to backend database and returns its information
+   * @param newWorkflowName
+   * @param newWorkflowCreationTime
+   * @param newWorkflowLastModifiedTime
+   * @param newWorkflowContent
+   */
+  public createWorkflow(newWorkflowName: string, newWorkflowCreationTime: number|undefined, newWorkflowLastModifiedTime: number|undefined,
+                        newWorkflowContent: WorkflowContent): Observable<Workflow> {
+    return this.http.post<Workflow>(`${AppSettings.getApiEndpoint()}/${WORKFLOW_URL}/create`, {
+      wid: undefined,
+      name: newWorkflowName,
+      creationTime: newWorkflowCreationTime,
+      lastModifiedTime: newWorkflowCreationTime,
+      content: JSON.stringify(newWorkflowContent)
+    })
```

Comment:
Use object Workflow as the argument to post.

Reply:
I'm confused about this one too, If I don't pass `Workflow: workflow` in the function's parameter, how am I able to have an object Workflow as the argument to post?
Is it to initialize a new workflow object upon the return statement? But Workflow is an interface.
I was passing the post argument in this format for the reference of the code written in the function `persistWorkflow(workflow: Workflow)`

BTW, since we don't need creationTime and lastModifiedTime, now my version of code look like this:
`wid: workflow.wid,
 name: workflow.name,
 content: JSON.stringify(workflow.content)`

I have little experience on this, so it might be stupid question, feel free to correct me.

Reply:
now since we removed some of the arguments, we can keep it like this